station-blocks-v0/src/main/java/net/modificationstation/stationapi/mixin/block/client/WorldRendererMixin.java

```diff
@@ -0,0 +1,76 @@
+package net.modificationstation.stationapi.mixin.block.client;
+
+import net.minecraft.block.Block;
+import net.minecraft.client.render.Tessellator;
+import net.minecraft.client.render.WorldRenderer;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.hit.HitResult;
+import net.minecraft.util.hit.HitResultType;
+import net.minecraft.world.World;
+import net.modificationstation.stationapi.api.block.BlockState;
+import net.modificationstation.stationapi.api.block.HasVoxelShape;
+import net.modificationstation.stationapi.api.util.math.Vec3d;
+import net.modificationstation.stationapi.impl.block.Line;
+import org.lwjgl.opengl.GL11;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import java.util.List;
+
+import static net.modificationstation.stationapi.impl.block.BoxToLinesConverter.convertBoxesToLines;
+
+@Mixin(WorldRenderer.class)
+public abstract class WorldRendererMixin {
+    @Shadow private World world;
+
+    @Unique
+    Block lastBlock = null;
+    @Unique
+    HitResult lastHitResult = null;
+    @Unique
+    List<Line> lines = null;
+
+    @Inject(method = "method_1554(Lnet/minecraft/entity/player/PlayerEntity;Lnet/minecraft/util/hit/HitResult;ILnet/minecraft/item/ItemStack;F)V", at = @At(value = "INVOKE", target = "Lnet/minecraft/world/World;getBlockId(III)I"), cancellable = true)
```

Comment:
Consider a less invasive injection. There's some vanilla code copying of which can be avoided.